libs/auth-api-lib/src/lib/services/delegation-scope.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Logger, LOGGER_PROVIDER } from '@island.is/logging'
+import { Inject, Injectable } from '@nestjs/common'
+import { InjectModel } from '@nestjs/sequelize'
+import { DelegationScopeDTO } from '../entities/dto/delegation-scope.dto'
+import { DelegationScope } from '../entities/models/delegation-scope.model'
+
+@Injectable()
+export class DelegationScopeService {
+  constructor(
+    @InjectModel(DelegationScope)
+    private delegationScopeModel: typeof DelegationScope,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) {}
+
+  async create(
+    delegationScope: DelegationScopeDTO,
+  ): Promise<DelegationScope | null> {
+    this.logger.debug('Creating new delegation scope')
+    return await this.delegationScopeModel.create({ ...delegationScope })
+  }
+
+  async createMany(delegationId: string, scopes: string[]): Promise<any> {
+    const arr: DelegationScopeDTO[] = []
+    for (let i = 0; i < scopes.length; i++) {
+      arr.push({ delegationId: delegationId, scopeName: scopes[i] })
+    }
```

Comment:
Just a food for thought: using a `map` is most often preferred due to its immutability and readability (the code becomes more predictable):
```typescript
const delegationScopes: DelegationScopeDTO[] = scopes.map(scope => ({delegationId, scopeName: scope}))
```